slacker/slack_api.py

```diff
@@ -66,9 +69,19 @@ def post(self, method, args={}):
     if self.__requires_token:
       args["token"] = self.__token
 
+    # Check for cached request
+    cache_key = self.__generate_cache_key(url, args)
+    if cache_key is not None and self.__is_cached(cache_key):
+      cache_value = self.__get_cached_value(cache_key)
+      if cache_value is not None:
+        return cache_value
```

Comment:
Look-ups are made twice here. Instead, you could could do one look-up with `self.__get_cached_value` and use its return value as evidence of being cached or not. Afterwards, you can then simply use the value and no more look-ups are necessary.